analyzer/src/main/kotlin/managers/Conan.kt

```diff
@@ -144,25 +142,47 @@ class Conan(
         stashDirectories(directoryToStash).use {
             configureRemoteAuthentication(conanConfig)
 
+            // TODO: Support lockfiles which are located in a different directory than the definition file.
+            val lockfileName = analyzerConfig.packageManagers?.get(managerName)?.options?.get("lockfileName")
+            requireLockfile(workingDir) { lockfileName?.let { hasLockFile(workingDir.resolve(it).path) } ?: false }
+
             val jsonFile = createOrtTempDir().resolve("info.json")
             run(workingDir, "info", definitionFile.name, "--json", jsonFile.absolutePath, *DUMMY_COMPILER_SETTINGS)
 
             val pkgInfos = jsonMapper.readTree(jsonFile)
             jsonFile.parentFile.safeDeleteRecursively(force = true)
 
-            val packageList = removeProjectPackage(pkgInfos, definitionFile.name)
+            val filteredPkgInfos = if (!analyzerConfig.allowDynamicVersions && lockfileName != null) {
+                val lockfileNodes = jsonMapper.readTree(definitionFile.resolveSibling(lockfileName))
+                val lockfilePackages = lockfileNodes["graph_lock"]["nodes"].mapNotNull {
+                    it["ref"]?.textValue()
+                }
+
+                jsonMapper.createArrayNode().addAll(
+                    pkgInfos.filter { jsonNode ->
+                        lockfilePackages.contains(jsonNode["reference"].textValueOrEmpty())
+                                || jsonNode["reference"].textValueOrEmpty().contains(definitionFile.name)
+                    }
```

Comment:
Can you explain why this was implemented as a filter instead of using the `--lockfile` option of the `conan info` command?

Reply:
@mnonnenmacher you are right. I must have overlooked this because when I was testing conan with a complex example project that had a lot of interconnected dependencies, it gave me an impression that `-l` option is used to create the lockfile instead of reading it. I will make the change to this PR.

Reply:
@mnonnenmacher the change has been done